strawman-Array.prototype.last.md

```diff
@@ -0,0 +1,102 @@
+# Array.prototype.last
+A proposal for an ECMAScript native property
+
+Code
+---
+
+    //sample code
+    Object.defineProperty(Array.prototype, 'last', {
+        configurable: true,
+        get         : function () {
+            if (this === null || typeof this.length !== "number") {
+                //verify if reference is still an array
+                throw new TypeError;
+            }
+            if (this.length < 0 || this.length > 2147483647) {
+                //checking for inclusive range of 0 to 2^31
+                //2^31 is the 2gb numeric threshold
+                throw new RangeError;
+            }
+            if (this.length === 0) {
+                //undefined would be returned anyways, but this condition
+                //specifies such as intentional behavior and not an oversight
```

Comment:
I do not think that is correct behavior even if it is allowed by some browsers. I have observed identical behavior across Firefox, Chrome, and Safari for assigning to a negative index:

* The array returns `[]`, an empty array.
* The array length returns 0, which is reflective of an empty array.
* The array `pop()` method does not remove the index.
* The array does complete the assignment and retain the value to the chosen index.

This behavior, to me, indicates assignment via array notation to an object literal. If you exchange the index value of `-1` for any string the behavior is identical. I think this is an unnatural feature that results from the array type really being identical to the object type but with access to Array.prototype.

Do you believe `void 0` is safer because `undefined` is capable of being assigned a value?